gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/utils/ThymeleafDataHelper.java

```diff
@@ -72,34 +72,43 @@ public static Map<String, Object> generateData(RoutingContext context, Domain do
         }
         data.put(PARAM_CONTEXT_KEY, parameters);
         if (context.session() != null && context.getDelegate() != null) {
+            MultiMap queryParams = context.queryParams();
             String errorHash = context.session().get(ERROR_HASH);
+            String error = queryParams.get(ERROR_PARAM_KEY);
+            String errorCode = queryParams.get(ERROR_CODE_PARAM_KEY);
+            String errorDescription = queryParams.get(ERROR_DESCRIPTION_PARAM_KEY);
             if (errorHash != null) {
-                StringBuilder error = new StringBuilder();
-                MultiMap queryParams = context.queryParams();
-                if (queryParams.get(ERROR_PARAM_KEY) != null) {
+                StringBuilder errorSB = new StringBuilder();
+                if (error != null) {
                     data.put(ERROR_PARAM_KEY, queryParams.get(ERROR_PARAM_KEY));
-                    error.append(queryParams.get(ERROR_PARAM_KEY));
+                    errorSB.append(queryParams.get(ERROR_PARAM_KEY));
                 }
-                if (queryParams.get(ERROR_CODE_PARAM_KEY) != null) {
+                if (errorCode != null) {
                     data.put(ERROR_CODE_PARAM_KEY, queryParams.get(ERROR_CODE_PARAM_KEY));
                 }
-                if (queryParams.get(ERROR_DESCRIPTION_PARAM_KEY) != null) {
+                if (errorDescription != null) {
                     data.put(ERROR_DESCRIPTION_PARAM_KEY, queryParams.get(ERROR_DESCRIPTION_PARAM_KEY));
-                    error.append("$");
-                    error.append(queryParams.get(ERROR_DESCRIPTION_PARAM_KEY));
+                    errorSB.append("$");
+                    errorSB.append(queryParams.get(ERROR_DESCRIPTION_PARAM_KEY));
                 }
-                if (!StringUtil.isBlank(error.toString()) && !HashUtil.compare(errorHash, error.toString())) {
-                    data.put(ERROR_PARAM_KEY, SERVER_ERROR);
-                    data.put(ERROR_CODE_PARAM_KEY, null);
-                    data.put(ERROR_DESCRIPTION_PARAM_KEY, "Unknown error occurred");
+                if (!StringUtil.isBlank(errorSB.toString()) && !HashUtil.compare(errorHash, errorSB.toString())) {
+                    setServerError(data);
                 }
+            } else if (error != null || errorCode != null || errorDescription != null) {
+                setServerError(data);
             }
```

Comment:
question: is it something we want to do even if the session is null ? Here the condition to call the setServerError method is the session existance but what is the session is null ? :thinking:

Reply:
discussed this point internally, seems ok